src/widgetastic_patternfly4/bulletchart.py

```diff
@@ -0,0 +1,161 @@
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import Text
+from widgetastic.widget import Widget
+from widgetastic.xpath import quote
+
+
+class Legend:
+    """Represents Legend of chart."""
+
+    def __init__(self, label, value=None, color=None, element=None):
+        self.label = label
+        self.value = value
+        self.color = color
+        self.element = element
+
+    def click(self):
+        self.element.click()
+
+    def __repr__(self):
+        return f"Legend({self.label}: {self.value})"
+
+    def __gt__(self, leg):
+        return self.__class__ == leg.__class__ and self.value > leg.value
+
+    def __eq__(self, leg):
+        return (
+            self.__class__ == self.__class__ and self.label == leg.label and self.value == leg.value
+        )
+
+
+class Bar(Legend):
+    """Represents Bar of chart."""
+
+    def __repr__(self):
+        return f"Bar({self.label}: {self.value})"
+
+
+class BulletChart(Widget):
+    """Represents the Patternfly Bullet Chart.
+
+    https://www.patternfly.org/v4/documentation/react/charts/chartbullet
+
+    Args:
+        id: If you want to look the input up by id, use this parameter, pass the id.
+        locator: If you have specific locator else it will take pf-chart.
+        offset_denominator: Denominator for offset value calculation.
+    """
+
+    ROOT = ParametrizedLocator("{@locator}")
+    DEFAULT_LOCATOR = ".//div[contains(@class, 'chartbullet')]"
+    LEGEND_ICON = (
+        ".//*[name()='g' or name()='svg']/*[name()='rect']/following-sibling::*[name()='path']"
+    )
+    LEGEND_TEXT = ".//*[name()='text' and contains(@id, 'legend-labels')]"
+    LEGEND_ITEM_REGEX = re.compile(r"(\d+)\s(\w.*)|(\w.*)\s(\d+)")
+
+    ITEMS = ".//*[name()='g']/*[name()='path' and not(contains(@style, 'type:square'))]"
+    TOOLTIP_REGEX = re.compile(r"(.*?): ([\d]+)")
+
+    tooltip = Text(
+        ".//*[name()='svg' and contains(@aria-labelledby, 'victory-container')]/"
+        "following-sibling::div[contains(@style, 'z-index')]/*[name()='svg']"
+    )
+
+    def __init__(self, parent=None, id=None, locator=None, logger=None, *args, **kwargs):
+        Widget.__init__(self, parent=parent, logger=logger)
+        if id:
+            self.locator = ".//div[@id={}]".format(quote(id))
+        elif locator:
+            self.locator = locator
+        else:
+            self.locator = self.DEFAULT_LOCATOR
+
+        self.args = args
+        self.kwargs = kwargs
+
+    def _offsets(self, el):
+        """Calculate offset. Need to set offset with try and error method."""
+        offset_denominator = self.kwargs.pop("offset_denominator", 2.5)
+        width = el.size["width"]
+        height = el.size["height"]
+        dx = int(width / offset_denominator) if (width > 10 and height > 10) else 0
+        dy = int(height / offset_denominator) if (width > 10 and height > 10) else 0
+        return dx, dy
+
+    @property
+    def legends(self):
```

Comment:
A good candidate for `@ParametrizedView.nested`

Reply:
I thought same with first look but after inspect I found its difficult to set `ROOT` as I want to consider legend `icon` and `text` and then decided to iterate over zip.

![Screenshot_2020-06-15_19-01-24](https://user-images.githubusercontent.com/11618054/84663414-c36ebd80-af3a-11ea-82bc-25accf3c0ba2.png)